### PR TITLE
Filestore space cli deleters

### LINKED ISSUE #1155: Add a space-cli delete <resource> command
### The problem faced currently?
Currently we can apply config to SpaceCloud. This solves the need to create and update resources. However, there is no mechanism to delete a resource via. space-cli.
### How can we solve it?
We can add `deleters` for each resource. This will be similar to the `generators` and `getters`. We will have to be careful to make sure we give flexibility to delete resources by `id`, `prefix` (this can be a multi select) or delete all resources.
>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by KavishShah09**
fixes #1155 for filestore module

### CODE REVIEW

File: gateway/managers/syncman/syncman_file.go [hunk 1]
```diff
@@ -36,6 +36,31 @@ func (s *Manager) SetFileStore(ctx context.Context, project string, value *confi
...
+		return http.StatusInternalServerError, helpers.Logger.LogError(helpers.GetRequestID(ctx), "error setting file store config", err, nil)
+	}
+
+	resourceID := config.GenerateResourceID(s.clusterID, project, config.ResourceFileStoreConfig, "filestore")
+	if err := s.store.DeleteResource(ctx, resourceID); err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	return http.StatusOK, nil
+}
```
Comment (sharadregoti) [inline gateway/managers/syncman/syncman_file.go:62]: I don't think so this is required, you just call the set file store config endpoint with all the required values in the payload as empty

File: space-cli/cmd/modules/filestore/commands.go [hunk 2]
```diff
@@ -60,6 +45,27 @@ func GetSubCommands() []*cobra.Command {
...
 
+// DeleteSubCommands is the list of commands the filestore module exposes
+func DeleteSubCommands() []*cobra.Command {
+
+	var deleteFileStoreRules = &cobra.Command{
+		Use:               "filestore-rules",
+		Aliases:           []string{"filestore-rule"},
+		RunE:              actionDeleteFileStoreRule,
+		ValidArgsFunction: fileStoreRulesAutoCompleteFun,
+		Example:           "space-cli delete filestore-rules ruleID --project myproject --log-level info",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/filestore/commands.go:56]: Remove --log level info from usage.
Do this everywhere

File: space-cli/cmd/utils/filter/filter.go [hunk 3]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 // DeleteOptions filters the resource id based on prefix to delete
-func DeleteOptions(project, prefix string, resources []string, doesResourceExist bool) (string, error) {
+func DeleteOptions(prefix string, resources []string, doesResourceExist bool) (string, error) {
```
Comment (sharadregoti) [inline space-cli/cmd/utils/filter/filter.go:12]: chnage line number 15 to
		if prefix != "" && strings.HasPrefix(strings.ToLower(resource), strings.ToLower(prefix)) && prefix != resource{

And remove the if at line number 22, always show the filtered resources

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Comment by KavishShah09** [comment]
@sharadregoti please add `hacktoberfest-accepted` label to this

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**